backend/src/BuiltinExecution/Libs/List.fs

```diff
@@ -665,15 +674,16 @@ let fns : List<BuiltInFn> =
          selected value in <param list>. Returns {{None}} if <param list> is
          empty."
       fn =
+        let optType = VT.unknownTODO
```

Comment:
Is optType not the same as the first param in the DList?

Reply:
I think you're implying that we could update the implementation of List.randomElement to something like:

```fsharp
let optType = VT.unknownTODO
        (function
        | _, _, [ DList(_, []) ] -> Ply(Dval.optionNone optType)
        | _, _, [ DList(_, l) ] ->
          // Will return <= (length - 1)
          // Maximum value is Int64.MaxValue which is half of UInt64.MaxValue, but
          // that won't affect this as we won't have a list that big for a long long
          // long time.
          let index = RNG.GetInt32(l.Length)
          match List.tryItem index l with
          | Some item -> Dval.optionSome (Dval.toValueType item) item |> Ply
          | None -> Ply(Dval.optionNone optType)
```
?